challenge_13/python/ning/challenge_13.py

```diff
@@ -1,54 +1,64 @@
-def get_int_len(_int):
+def get_len(_int):
     '''Find the number of digits of an int
+    if not mathematical integer (1.0 allowed)
+    (e.g. 1.1), return -1
 
     Given an integer, finds the number of
     digits in that integer in base 10
     '''
-    power = 1
+    power = 0
+
+    if _int % 1 != 0:
+        return -1
+
     while True:
-        if _int == 0:
-            return 0
-        elif _int < 10**power:
+        if _int < 10**power:
             return power
         power += 1
 
-def check_palindrome(_int):
-    ones = 0
-    digits = get_int_len(_int)
-    while digits > 1:
-        if _int % 10 != 0:  # not end with 0
-            ones += 1
-            _int -= 1
-        else:
-            _int -= ones * 10**(digits-1)
-            _int /= 10
-            new_digits = get_int_len(_int)
-            diff_digits = digits - new_digits
-            if diff_digits == 1 or _int < 0:
-                # e.g. 514, 510, 110, 10
-                # e.g. 415, 410, -90, -9
-                return False
-            elif diff_digits > 2:
-                # e.g. 10101, 10100, 0010
-                extra_zeroes = diff_digits - 2
-                _int /= 10**(extra_zeroes)
-                if _int % 1 != 0:
-                    return False
-            digits = get_int_len(_int)
-            ones = 0
-
-    return True
-
-def check_palindrome_str(_str):
-    return check_palindrome(int(_str))
+def is_palindrome(_int):
+    len_pre = get_len(_int)
+
+    # base cases
+    if len_pre == -1:
+        # e.g. 4323.4 (refer func get_len)
+        return False
+    elif len_pre == 0 or len_pre == 1:
+        # e.g. 0, 1, 2, ...
+        return True
+    elif len_pre == 2:
+        # e.g. 11, 22, 33, ...
+        return _int % 11 == 0
+
+    # recursive reduction
+    else:
+        digit_last = _int % 10
+        _int -= digit_last
```

Comment:
I agree. Kicked myself when I saw the other _way_ simpler solutions to this challenge. This solution logic was merged, however, in #421. This pull request is actually just an attempt on an improvement on the last using the same logic, but improving on code style (this was recommended to me by @slandau3 over slack PM).